### ISSUE TITLE
local-up-cluster.sh not playing nice with ~/.kube/config - causing login error

### ISSUE THREAD

**Issue by screeley44** [kubernetes/kubernetes#40459]
using ./hack/local-up-cluster.sh and after executing the post completion steps get a login failure:

```
[root@k8dev2 kubernetes]# kubectl get nodes
error: You must be logged in to the server (the server has asked for the client to provide credentials (get .meta.k8s.io))
```

Post Completion Steps:
```
To start using your cluster, open up another terminal/tab and run:

  export KUBERNETES_PROVIDER=local

  cluster/kubectl.sh config set-cluster local --server=https://localhost:6443 --certificate-authority=/var/run/kubernetes/apiserver.crt
  cluster/kubectl.sh config set-credentials myself --username=admin --password=admin
  cluster/kubectl.sh config set-context local --cluster=local --user=myself
  cluster/kubectl.sh config use-context local
  cluster/kubectl.sh
```

After speaking with @ncdc he asked me to try the following:

1.  remove the ~/.kube/config  file
2.  remove /var/run/kubernetes
3.  rerun local-up-cluster.sh

After those steps, IF I DO NOT run the post script steps from above where we are setting the config...
I'm able to use kubectl with no issues

```
[root@k8dev2 kubernetes]# kubectl get nodes
NAME        STATUS    AGE       VERSION
127.0.0.1   Ready     10s       v1.6.0-alpha.0.3907+90b5d4cbd8719f
```

However, as soon as I run those steps, I'm back to the error:

```
Cluster "local" set.
User "myself" set.
Context "local" set.
Switched to context "local".

[root@k8dev2 kubernetes]# kubectl get nodes
error: You must be logged in to the server (the server has asked for the client to provide credentials (get .meta.k8s.io))
```

After this, without restarting my cluster, I just simply remove the ~/.kube/config file again and I'm back to a working cluster:

```
[root@k8dev2 kubernetes]# rm -rf ~/.kube/config 

[root@k8dev2 kubernetes]# kubectl get nodes
NAME        STATUS    AGE       VERSION
127.0.0.1   Ready     14m       v1.6.0-alpha.0.3907+90b5d4cbd8719f
```

**Comment by screeley44** [comment]
cc @deads2k

**Comment by PI-Victor** [comment]
fwiw - valid, i'm hitting the same issue. removing the kube folder seems to make it work.

**Comment by tonglil** [comment]
Same problem.
If I remove the `~/.kube/config` file, it shows no nodes.
Does this script only work on Linux?

**Comment by gyliu513** [comment]
@tonglil I have the same issue and after remove `~/.kube/config`, the cluster works well and I can also get all nodes.

```
root@k8s001:~/go/src/k8s.io/kubernetes# kubectl get nodes
NAME        STATUS    AGE
127.0.0.1   Ready     7m
```

**Comment by MHBauer** [comment]
Leaving the kubeconfig blank, or deleting `~/.kube/config`, or unsetting the context with `kubectl config unset current-context` results in the client using an insecure connection.

**Comment by tonglil** [comment]
@gyliu513 when I remove/move `~/.kube/config` I get:
```
$ kubectl get nodes
No resources found.
```

**Comment by tonglil** [comment]
I manually updated `hack/local-up-cluster.sh` and added `--anonymous-auth=true` to the api server and now it works.

**Comment by tonglil** [comment]
Is anyone running on OS X? I wonder if the warning means I can't put up a local cluster on OS X:
```
...
2017/02/09 14:55:29 [INFO] generate received request
2017/02/09 14:55:29 [INFO] received CSR
2017/02/09 14:55:29 [INFO] generating key: rsa-2048
2017/02/09 14:55:30 [INFO] encoded CSR
2017/02/09 14:55:30 [INFO] signed certificate with serial number 22921173938288349138325629150385695161911363743
2017/02/09 14:55:30 [WARNING] This certificate lacks a "hosts" field. This makes it unsuitable for
websites. For more information see the Baseline Requirements for the Issuance and Management
of Publicly-Trusted Certificates, v.1.1.6, from the CA/Browser Forum (https://cabforum.org);
specifically, section 10.2.3 ("Information Requirements").
Waiting for apiserver to come up
+++ [0209 14:55:31] On try 2, apiserver: : {
  "major": "1",
  "minor": "7+",
  "gitVersion": "v1.7.0-923424afa0.1+3b18a355e87915-dirty",
  "gitCommit": "3b18a355e87915124fa7cdd0bc48513fe8601546",
  "gitTreeState": "dirty",
  "buildDate": "2017-02-09T19:44:35Z",
  "goVersion": "go1.7.5",
  "compiler": "gc",
  "platform": "darwin/amd64"
}

WARNING: kubelet is not currently supported in darwin, kubelet aborted.

Local Kubernetes cluster is running. Press Ctrl-C to shut it down.

Logs:
  /tmp/kube-apiserver.log
  /tmp/kube-controller-manager.log
  /tmp/kube-proxy.log
  /tmp/kube-scheduler.log
  

To start using your cluster, open up another terminal/tab and run:

  export KUBERNETES_PROVIDER=local

  cluster/kubectl.sh config set-cluster local --server=https://localhost:6443 --certificate-authority=/var/run/kubernetes/apiserver.crt
  cluster/kubectl.sh config set-credentials myself --username=admin --password=admin
  cluster/kubectl.sh config set-context local --cluster=local --user=myself
  cluster/kubectl.sh config use-context local
  cluster/kubectl.sh
```

**Comment by ncdc** [comment]
I don't believe this is an issue any more. The latest commit does this:

```
To start using your cluster, you can open up another terminal/tab and run:

  export KUBECONFIG=/var/run/kubernetes/admin.kubeconfig
  cluster/kubectl.sh

Alternatively, you can write to the default kubeconfig:

  export KUBERNETES_PROVIDER=local

  cluster/kubectl.sh config set-cluster local --server=https://localhost:6443 --certificate-authority=/var/run/kubernetes/apiserver.crt
  cluster/kubectl.sh config set-credentials myself --client-key=/var/run/kubernetes/client-admin.key --client-certificate=/var/run/kubernetes/client-admin.crt
  cluster/kubectl.sh config set-context local --cluster=local --user=myself
  cluster/kubectl.sh config use-context local
  cluster/kubectl.sh
```

The only "issue" I have is that I have to `chown` the files in `/var/run/kubernetes` to my user since they're owned by root.

**Comment by MHBauer** [comment]
@ncdc yes. this works for me for this issue.

**Comment by MHBauer** [comment]
fixed by #40193
